File: api-tsl/src/tr/gov/tubitak/uekae/esya/api/tsl/model/schemeinformation/TSLServiceDigitalIdentity.java
package tr.gov.tubitak.uekae.esya.api.tsl.model.schemeinformation;

import org.w3c.dom.Document;
import org.w3c.dom.Element;

import tr.gov.tubitak.uekae.esya.api.asn.x509.ECertificate;
import tr.gov.tubitak.uekae.esya.api.common.ESYAException;
import tr.gov.tubitak.uekae.esya.api.tsl.TSLException;
import tr.gov.tubitak.uekae.esya.api.tsl.model.core.BaseElement;
import tr.gov.tubitak.uekae.esya.api.tsl.util.Base64;
import tr.gov.tubitak.uekae.esya.api.tsl.util.Constants;
import tr.gov.tubitak.uekae.esya.api.tsl.util.XmlUtil;

public class TSLServiceDigitalIdentity extends BaseElement {
	private String certificateBase64Str;
	private ECertificate certificate;

	private Element digitalIdElement;
	private Element x509CertificateElement;

	public TSLServiceDigitalIdentity(Document document,
			ECertificate iCertificate) {
		super(document);
		certificate = iCertificate;
		 certificateBase64Str = Base64.encode(certificate.getEncoded());

		x509CertificateElement = createElement(Constants.NS_TSL,
				Constants.TAG_X509CERTIFICATE);
		x509CertificateElement.setTextContent(certificateBase64Str);

		digitalIdElement = createElement(Constants.NS_TSL,
				Constants.TAG_DIGITALID);
		digitalIdElement.appendChild(document.createTextNode("\n"));
		digitalIdElement.appendChild(x509CertificateElement);
		digitalIdElement.appendChild(document.createTextNode("\n"));

		addLineBreak();
		mElement.appendChild(digitalIdElement);
		addLineBreak();
	}

	public TSLServiceDigitalIdentity(Element aElement) throws TSLException, ESYAException {
		super(aElement);
		digitalIdElement = XmlUtil.getNextElement(aElement.getFirstChild());
		x509CertificateElement = XmlUtil.getNextElement(digitalIdElement
				.getFirstChild());

		certificateBase64Str = XmlUtil.getText(x509CertificateElement);
		 certificate = new ECertificate(certificateBase64Str);
	}

	@Override
	public String getLocalName() {
		return Constants.TAG_SERVICEDIGITALIDENTITY;
	}

	@Override
	public String getNamespace() {
		return Constants.NS_TSL;
	}

	public ECertificate getX509Certificate() {
		return certificate;
	}

	public String getBase64EncodedCertificate() {
		return certificateBase64Str;
	}
}

File: api-tsl/csharp/src/model/schemeinformation/TSLServiceDigitalIdentity.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Xml;
using tr.gov.tubitak.uekae.esya.api.tsl.model.core;
using tr.gov.tubitak.uekae.esya.api.tsl.util;
using tr.gov.tubitak.uekae.esya.api.asn.x509;
using tr.gov.tubitak.uekae.esya.api.common.util;

namespace tr.gov.tubitak.uekae.esya.api.tsl.model.schemeinformation
{
    public class TSLServiceDigitalIdentity : BaseElement
    {
        private readonly string certificateBase64Str;
        private readonly ECertificate certificate;
        
        private readonly XmlElement digitalIdElement;
        private readonly XmlElement x509CertificateElement;

        public TSLServiceDigitalIdentity(XmlDocument document, ECertificate iCertificate) : base(document)
        {
            certificate = iCertificate;
            certificateBase64Str = Convert.ToBase64String(certificate.getBytes());

            x509CertificateElement = createElement(Constants.NS_TSL,Constants.TAG_X509CERTIFICATE);
            x509CertificateElement.InnerText = certificateBase64Str;

            digitalIdElement = createElement(Constants.NS_TSL,Constants.TAG_DIGITALID);
            digitalIdElement.AppendChild(document.CreateTextNode("\n"));
            digitalIdElement.AppendChild(x509CertificateElement);
            digitalIdElement.AppendChild(document.CreateTextNode("\n"));
           
            addLineBreak();
            mElement.AppendChild(digitalIdElement);
            addLineBreak();
        }

        public TSLServiceDigitalIdentity(XmlElement aElement) : base(aElement)
        {
            digitalIdElement = XmlUtil.getNextElement(aElement.FirstChild);
            x509CertificateElement = XmlUtil.getNextElement(digitalIdElement.FirstChild);

            certificateBase64Str = XmlUtil.getText(x509CertificateElement);
            certificate = new ECertificate(certificateBase64Str);
        }

        public override string LocalName
        {
            get { return Constants.TAG_SERVICEDIGITALIDENTITY; }
        }

        public override string Namespace
        {
            get
            {
                return Constants.NS_TSL;
            }
        }

        public ECertificate X509Certificate
        {
            get { return certificate; }
        }

        public string Base64EncodedCertificate
        {
            get { return certificateBase64Str; }
        }
    }
}
